packages/react/toast/src/Toast.tsx

```diff
@@ -805,18 +809,43 @@ const ToastAnnounceExclude = React.forwardRef<
   ToastAnnounceExcludeElement,
   ToastAnnounceExcludeProps
 >((props: ScopedProps<ToastAnnounceExcludeProps>, forwardedRef) => {
-  const { __scopeToast, altText, ...announceHiddenProps } = props;
+  const { __scopeToast, altText, ...announceExcludeProps } = props;
 
   return (
     <Primitive.div
-      data-radix-toast-announce-exclude={altText ? undefined : ''}
+      data-radix-toast-announce-exclude=""
       data-radix-toast-announce-alt={altText || undefined}
-      {...announceHiddenProps}
+      {...announceExcludeProps}
       ref={forwardedRef}
     />
   );
 });
 
+function getAnnounceTextContent(container: HTMLElement) {
+  const textContent: string[] = [];
+  const childNodes = Array.from(container.childNodes);
+
+  childNodes.forEach((node) => {
+    if (node.nodeType === node.TEXT_NODE && node.textContent) textContent.push(node.textContent);
+    if (isHTMLElement(node)) {
+      const isHidden = node.ariaHidden || node.hidden || node.style.display === 'none';
+      const isExcluded = node.dataset.radixToastAnnounceExclude === '';
```

Comment:
nit (perhaps? untested): 
```suggestion
      const isExcluded = 'radixToastAnnounceExclude' in node.dataset;
```

Reply:
In this case, I guess we can keep this as is too then.